scratch/evaluate_qd_channel_lroom_scenario.cc

```diff
@@ -189,33 +191,63 @@ CongStateTrace (TcpSocketState::TcpCongState_t oldState, TcpSocketState::TcpCong
   NS_LOG_DEBUG ("Old State: " << oldState << ", New State: " << newState); 
 }
 
+DmgTspecElement
+GetDmgTspecElement (uint8_t allocId, bool isPseudoStatic, uint32_t minAllocation, uint32_t maxAllocation)
+{
+  /* Simple assert for the moment */
+  NS_ASSERT_MSG (minAllocation <= MAX_SP_BLOCK_DURATION, "Exceeding Max SP block duration");
+  NS_ASSERT_MSG (maxAllocation <= MAX_SP_BLOCK_DURATION, "Exceeding Max SP block duration");
+  DmgTspecElement element;
+  DmgAllocationInfo info;
+  info.SetAllocationID (allocId);
+  info.SetAllocationType (SERVICE_PERIOD_ALLOCATION);
+  info.SetAllocationFormat (ISOCHRONOUS);
+  info.SetAsPseudoStatic (isPseudoStatic);
+  info.SetDestinationAid (AID_AP);
+  element.SetDmgAllocationInfo (info);
+  element.SetMinimumAllocation (minAllocation);
+  element.SetMaximumAllocation (maxAllocation);
+  element.SetMinimumDuration (minAllocation);
+  return element;
+} 
+
 void
-StationAssociated (Ptr<DmgWifiMac> staWifiMac, Mac48Address address, uint16_t aid)
+StationAssociated (Ptr<DmgStaWifiMac> staWifiMac, Mac48Address address, uint16_t aid)
 {
   if (!csv)
     {
       std::cout << "DMG STA " << staWifiMac->GetAddress () << " associated with DMG PCP/AP " << address
                 << ", Association ID (AID) = " << aid << std::endl;
     }
-  appStartTime = Simulator::Now ();
-  if (applicationType == "onoff")
+    staWifiMac->CreateAllocation (GetDmgTspecElement (1, true, 10000, 10000));
+    //staWifiMac->CreateAllocation (GetDmgTspecElement (2, true, 10000, 10000));
+    //staWifiMac->CreateAllocation (GetDmgTspecElement (3, true, 10000, 10000));
+    //Simulator::Schedule (Seconds (1.0), &DmgStaWifiMac::CreateAllocation, staWifiMac, GetDmgTspecElement (1, true, 30000, 30000));
+    //Simulator::Schedule (Seconds (1.0), &DmgStaWifiMac::CreateAllocation, staWifiMac, GetDmgTspecElement (3, true, 20000, 20000));
+    //Simulator::Schedule (Seconds (1.5), &DmgStaWifiMac::CreateAllocation, staWifiMac, GetDmgTspecElement (4, true, MAX_SP_BLOCK_DURATION, MAX_SP_BLOCK_DURATION));
+}
+
+void
+ADDTSResponseReceived (Mac48Address address, StatusCode status, DmgTspecElement element)
+{
+  NS_LOG_DEBUG (address << " Received ADDTS response with status: " << status.IsSuccess ());
+  if (status.IsSuccess () && !appStarted)
     {
-      onoff->StartApplication ();
-      /* Connect to TCP traces */
-      if (socketType == "ns3::TcpSocketFactory")
+      appStartTime = Simulator::Now ();
```

Comment:
What happens if we have multiple apps (even only two) running on the same device? Does the code remain consistent?

Reply:
No, one would need to modify the evaluation script accordingly.